README_for_direct_archiver.md

```diff
@@ -11,49 +11,74 @@ Follow the steps:
 
 3. Trigger the data archive request
-```
-curl -X POST {ingest_archiver_url}/archiveSubmissions/data -H 'Content-Type: application/json' -H "Api-Key:{archiver_api_key}" -d '{"sub_uuid": "{sub_uuid}"}'
-```
+   ```
+   curl -X POST {ingest_archiver_url}/archiveSubmissions/data -H 'Content-Type: application/json' -H "Api-Key:{archiver_api_key}" -d '{"sub_uuid": "{sub_uuid}"}'
+   ```
+
+   Where `sub_uuid` refers to the submission UUID for which sequence data files are to be archived.
 
-Where `sub_uuid` refers to the submission UUID for which sequence data files are to be archived.
+4. Check the data archive result. You might have to wait 1-2 minutes to get a proper result.
 
-4. Check the data archive result.
+   ```
+   curl -X GET {ingest_archiver_url}/archiveSubmissions/data/<sub_uuid> -H "Api-Key:{archiver_api_key}" 
+   ```
+   If the `fileArchiveResult` key is `null` in any of the `files` in the response, then probably you have to wait a bit more
+and repeat the `GET` request. If that does not help, then please go back to step 3
+and trigger the data file archiving again.
 
-```
-curl -X GET {ingest_archiver_url}/archiveSubmissions/data/<sub_uuid>' -H "Api-Key:{archiver_api_key}" 
-```
 Only proceed to metadata archive if all data files are successful archived. This may take a while for large submission so please be patient and check regularly until all files are archived.
 
 ## Metadata archive
 
 
-1. There is a script to trigger the direct archiver under the repository's root folder: `submit_to_archives.py`.
+1. Clone this repository to your computer and cd to the repository's root folder.
+2. Create and active virtual environment in the repository's root folder.
+
+   ```
+   python -mvenv .venv
+   source .venv/bin/activate
+   ```
+
+3. Install all the requirements for the repository.
+
+   ```
+   pip install -r requirements.txt
+   ```
+
+4. There is a script to trigger the direct archiver under the repository's root folder: `submit_to_archives.py`. 
 It needs the `submission_UUID` as the input parameter.
 
    There is a mandatory configuration parameter, too.
 You need to setup an environment variable for `ARCHIVER_API_KEY`. You can get this value from AWS Secret Manager.
 It is under the `ingest/archiver/wrangler/secrets` secret storage.
 You can get the above value from AWS Secret Manager with this command line action for dev environment.
 
-   `aws --region us-east-1 secretsmanager get-secret-value --secret-id ingest/archiver/wrangler/secrets --query SecretString --output text | jq -jr .dev_archiver_api_key; echo -e`
+   **Note**: You probably have done this step already when you archived the data files.
+In that case you don't have to repeat it, as you already have that value. 
 
+   ```
+   aws --region us-east-1 secretsmanager get-secret-value --secret-id ingest/archiver/wrangler/secrets --query SecretString --output text | jq -jr .dev_archiver_api_key; echo -e
```

Comment:
Is it always dev here? or do we need to set the correct env?

Reply:
@amnonkhen If you read it a bit further you get answer to your question. 😅

Reply:
@ke4 Why not use an environment variable and make it simpler?

```bash
export ENVIRONMENT=dev # or staging or prod
aws --region us-east-1 secretsmanager get-secret-value --secret-id ingest/archiver/wrangler/secrets --query SecretString --output text | jq -jr .${ENVIRONMENT}_archiver_api_key; echo -e
```

Reply:
Changed